pkg/local/sorted_queue.go

```diff
@@ -0,0 +1,28 @@
+package local
+
+// A SortedQueue implements heap.Interface and holds strings.
+type SortedQueue []string
+
+func (pq *SortedQueue) Len() int { return len(*pq) }
+
+func (pq *SortedQueue) Less(i, j int) bool {
+	// We want Pop to give us the smallest
+	return (*pq)[i] < (*pq)[j]
+}
+
+func (pq *SortedQueue) Swap(i, j int) {
```

Comment:
As above, you could add
```suggestion
func (pq *SortedQueue) Top() string {
	return pq[0]
}

func (pq *SortedQueue) Swap(i, j int) {
```

package heap explicitly says that "The minimum element in the tree is the root, at index 0. "